drivers/w1/w1_ds2477_85_common.c

```diff
@@ -0,0 +1,294 @@
+		LOG_ERR("invalid reset response");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+int ds2477_85_reset_bus(const struct device *dev)
+{
+	const struct w1_ds2477_85_config *cfg = dev->config;
+	struct w1_ds2477_85_data *data = dev->data;
+	uint16_t delay_us = cfg->mode_timing[data->master_reg.od_active].t_reset;
+	uint8_t tx_data;
+	uint8_t rx_data;
+	int ret;
+
+	tx_data = data->master_reg.od_active ? BIT(3) : BIT(7);
+	ret = cfg->w1_script_cmd(dev, delay_us, SCRIPT_OW_RESET, &tx_data, 1, &rx_data, 1);
+
+	switch (ret) {
+	case RESULT_COMM_FAILURE:
+		/* presence not detected */
+		return 0;
+	case RESULT_SUCCESS:
+		/* at least 1 device present */
+		return 1;
+	default:
+		return -EIO;
+	}
+}
+
+int ds2477_85_read_bit(const struct device *dev)
+{
+	const struct w1_ds2477_85_config *cfg = dev->config;
+	struct w1_ds2477_85_data *data = dev->data;
+	uint16_t delay_us = cfg->mode_timing[data->master_reg.od_active].t_slot;
+	uint8_t rx_data;
+	int ret;
+
+	ret = cfg->w1_script_cmd(dev, delay_us, SCRIPT_OW_READ_BIT, NULL, 0, &rx_data, 1);
+	if (ret != RESULT_SUCCESS) {
+		return -EIO;
+	}
+
+	return (rx_data & BIT(5)) ? 1 : 0;
+}
+
+int ds2477_85_write_bit(const struct device *dev, const bool bit)
+{
+	const struct w1_ds2477_85_config *cfg = dev->config;
+	struct w1_ds2477_85_data *data = dev->data;
+	uint16_t delay_us = cfg->mode_timing[data->master_reg.od_active].t_slot;
+	uint8_t tx_data = (uint8_t)bit;
+	uint8_t rx_data;
+	int ret;
+
+	ret = cfg->w1_script_cmd(dev, delay_us, SCRIPT_OW_WRITE_BIT, &tx_data, 1, &rx_data, 1);
+	if (ret != RESULT_SUCCESS) {
+		return -EIO;
+	}
+
+	return 0;
+}
+
+int ds2477_85_read_byte(const struct device *dev)
+{
+	const struct w1_ds2477_85_config *cfg = dev->config;
+	struct w1_ds2477_85_data *data = dev->data;
+	uint16_t delay_us = cfg->mode_timing[data->master_reg.od_active].t_slot * 8;
+	uint8_t rx_data;
+	int ret;
+
+	ret = cfg->w1_script_cmd(dev, delay_us, SCRIPT_OW_READ_BYTE, NULL, 0, &rx_data, 1);
+	if (ret != RESULT_SUCCESS) {
+		return -EIO;
+	}
+
+	return rx_data;
+}
+
+int ds2477_85_write_byte(const struct device *dev, const uint8_t tx_byte)
+{
+	const struct w1_ds2477_85_config *cfg = dev->config;
+	struct w1_ds2477_85_data *data = dev->data;
+	uint16_t delay_us = cfg->mode_timing[data->master_reg.od_active].t_slot * 8;
+	uint8_t rx_data;
+	int ret;
+
+	ret = cfg->w1_script_cmd(dev, delay_us, SCRIPT_OW_WRITE_BYTE, &tx_byte, 1, &rx_data, 1);
+	if (ret != RESULT_SUCCESS) {
+		return -EIO;
+	}
+
+	return 0;
+}
+
+int ds2477_85_write_block(const struct device *dev, const uint8_t *buffer, size_t tx_len)
+{
+	const struct w1_ds2477_85_config *cfg = dev->config;
+	struct w1_ds2477_85_data *data = dev->data;
+	int w1_timing = cfg->mode_timing[data->master_reg.od_active].t_slot * 8 * tx_len;
+	uint8_t buf[3] = {CMD_WRITE_BLOCK, (tx_len + 1), 0};
+	struct i2c_msg tx_msg[2] = {
+		{.buf = buf, .len = 3, .flags = I2C_MSG_WRITE},
+		{.buf = (uint8_t *)buffer, .len = tx_len, .flags = (I2C_MSG_WRITE | I2C_MSG_STOP)}
```

Comment:
nits

```suggestion
		[0] = {.buf = buf, .len = 3, .flags = I2C_MSG_WRITE},
		[1] = {.buf = (uint8_t *)buffer, .len = tx_len, .flags = (I2C_MSG_WRITE | I2C_MSG_STOP)},
```

Reply:
Adding the index would bring the line past 100 chars, so I believe it wouldn't get more readable.
So if it's just a preference I'd leave, otherwise I'll change it of course.